### PR TITLE
documentation and linting fixes for public/js/ files. [WIP]

### LINKED ISSUE #1426: Documentation of all Simulator's Javascript code.
**Is your feature request related to a problem? Please describe.**
Documenting the code for better understanding the codebase so that it can be easily refactored afterward. 
**Describe the solution you'd like**
Using JSDocs guide for comments and then making HTML files for Documentation using `jsdoc` 
**Describe alternatives you've considered**
NA
**Additional context**
NA
**Are you working on this? (Yes/No)**
Yes

### PR DESCRIPTION
**Pull request by shubhankarsharma00**
Fixes #1426 

#### Describe the changes you have made in this PR -
Added comments in accordance with jsdocs guidelines. Fixed some eslint errors added comments to ignore some error which I will fix later during refactoring.

### Screenshots of the changes (If any) -
temporarily hosted automatically generated documentation at: 
https://shubhankarsharma00.github.io/out/index.html

Note: Please check **Allow edits from maintainers.** if you would like us to assist in the PR.

### CODE REVIEW

File: public/js/data.js [hunk 1]
```diff
@@ -426,123 +457,122 @@ function generateImageForOnline() {
...
         // Create new project - this part needs to be improved and optimised
-        var form = $("<form/>", {
+        const form = $('<form/>', {
             action: '/simulator/create_data',
-            method: "post"
+            method: 'post',
         });
         form.append(
-            $("<input>", {
+            $('<input>', {
```
[the remark this answers is not in the document]
Comment (shubhankarsharma00) [reply public/js/data.js:490]: This is hidden input field

**Comment by shubhankarsharma00** [comment]
Appveyor throws this error:
```error mkdirp@1.0.4: The engine "node" is incompatible with this module. Expected version ">=10". Got "8.17.0"```
But it builds fine on Circle CI. There must be some offending dependency for `mkdirp` but it still should be a warning (I think).
Any Ideas?

**Comment by himanshu010** [comment]
> Appveyor throws this error:
> `error mkdirp@1.0.4: The engine "node" is incompatible with this module. Expected version ">=10". Got "8.17.0"`
> But it builds fine on Circle CI and my PC. There must be some offending dependency for `mkdirp` but it still should be a warning and not an error (I think).
> Any Ideas?
> Edit: It may be because Circle Ci doesn't check for node 8.17.0

I think use of ignore engines will help in this case
`yarn install --ignore-engines`

**Comment by shubhankarsharma00** [comment]
> I think use of ignore engines will help in this case
> `yarn install --ignore-engines`
Yeah, but I'm not sure if this is the correct answer. I think the best method would be to use the jsdoc version that can be built by node 8. Correct me if I'm wrong.

**Pull request closed without merging**